PEP8 fixes

delirious-lettuce opened the pull request:
This PR fixes most of #191 but there are still a few lines > 79 characters and probably a few other minor things that I missed. Here is a summary of the changes:

* Fix typo: `succesfully` -> `successfully`
* PEP8 fixes:

    * Reformat (most) lines longer than 79 chars  …
    * Two blank lines above top-level definitions
    * Add proper spacing around operators/commas
    * Use `snake_case` for function/variable names
    * Comments start with `# `
    * Use triple double-quoted strings for docstrings
    * Remove redundant parentheses
    * Simplify chained comparisons
    * Group imports (stdlib, third-party, local)
    * Remove unused imports
    * Remove excess spaces
    * Use `with` statements to ensure closing of files
    * Simplify conditionals `while ... == True -> while ...`
    * Remove unused variables
    * Use dict literals instead of multiple statements
    * Use a def instead of assigning a name to a lambda
    * Remove excess blank lines from EOF

mbonsma commented:
Thank you @delirious-lettuce! :+1: 

It looks like one of the tests failing on this error: `E   ValueError: <function test_phage_finder at 0x7f40dffc5400> uses no argument 'input'` 

I'm not sure why that's happening - it looks like that argument was indeed renamed, but I don't know why the test is failing now. @QuLogic if you have a chance sometime, do you know what this error is about?

delirious-lettuce commented:
@mbonsma ,

I'll go take a look right now.

delirious-lettuce commented:
@mbonsma ,

It looks like I forgot to update the variable name inside of `parametrize`. I updated it from `input` to `phage_input` which now matches the function argument.

I changed this because `input` is a builtin function name and shadowing them can be problematic.

* https://docs.python.org/2/library/functions.html#input
* https://docs.python.org/3/library/functions.html#input

Hopefully that fixes the issue!

delirious-lettuce commented:
@mbonsma ,

Sorry about that, I should have caught it earlier.

QuLogic commented on parserscripts/pfam_db.py:
```diff
@@ -93,26 +102,29 @@ def getValues(url):
-        url = getTranslationURL(translation)
-        result = getValues(url)
+        print("Getting translation " + str(i + 1) + "/" + str(
+            len(translations)) + "...")
```
You've got parentheses already; break before `str`, and align with opening parenthesis.

delirious-lettuce replied:
Damn, another one. I think these two (`str` breaks) were caused by PyCharm's `Reformat File`. I agree with your statement and I will fix it ASAP.

QuLogic commented on parserscripts/phage.py:
```diff
@@ -1,36 +1,36 @@
-    # National Center for Biotechnology Information phage database
-    "NCBI": r"^ENA\|([^\|]+)\|[^\ ]+\ ([^,]+)[^$]*$",
+class Phage:
+    supported_databases = {
```
This is a constant, no? If so, should be upper case.

delirious-lettuce replied:
> [Constants are usually defined on a module level and written in all capital letters with underscores separating words.](http://pep8.org/#constants)

It isn't defined at the module level and it is also mutable. I can change it if you want but I'm bit hesitant.

QuLogic replied:
One can define constants on a class-level as well if the encapsulation makes sense. Nothing stops a constant from being a mutable object; it's naming it with the conventional ALL_CAPS that says "don't modify me!".

Actually, since it's all convention, nothing stops you from modifying a constant that contains an immutable value either. 😉

delirious-lettuce replied:
No problem, just wanted to check that it's what you guys wanted. Commit coming in now.

QuLogic commented on populate_casgenes.py:
```diff
@@ -1,5 +1,7 @@
+
 import numpy as np
+
 # !/usr/bin/env python3
```
This is broken (though not by you.)

delirious-lettuce replied:
Which part? Do you want me to remove the shebang?

QuLogic replied:
The shebang must be on the first line. The encoding also should also be on the first two or three lines or something like that.

This was always broken though.

QuLogic replied:
Is this file even marked executable? Maybe no-one attempts to run it directly anyway; confirm with @mbonsma.

mbonsma replied:
Um yeah it should be executable? It's one of the one-time populate the database scripts.

QuLogic commented on populate_casgenes.py:
```diff
@@ -18,7 +20,6 @@
 
 '''
```
Merge this docstring with the first one?

delirious-lettuce replied:
Sure, no problem.

QuLogic commented on populate_selftargets.py:
```diff
@@ -1,25 +1,28 @@
+import os
+
+
 # !/usr/bin/env python3
```
Also broken.

delirious-lettuce replied:
Remove shebang?

QuLogic commented on util/prunedict.py:
```diff
@@ -1,48 +1,48 @@
+    incomp_list = []
+    for loc_id in gen_dict:
+        locus = gen_dict[loc_id]
         if not set(locus.keys()) == {'Spacers', 'Start', 'RepeatSeq', 'Stop'}:
```
`!=`

delirious-lettuce replied:
Ah yes, I'll change it ASAP.

delirious-lettuce commented on parserscripts/addRepeatAndSpacersToDB.py:
```diff
@@ -1,66 +1,73 @@
+    conn = sqlite3.connect(db_name)
+    c = conn.cursor()
+    col_name = name + 'Sequence'
+    q = "INSERT INTO " + str(name) + " (" + str(
```
Oops, I agree that looks ugly.

delirious-lettuce commented on parserscripts/bac_info_parser.py:
```diff
@@ -1,4 +1,4 @@
-#parse_Genbank.py
+# parse_Genbank.py
```
Ok, I'll delete it.

delirious-lettuce commented on parserscripts/crispr_db_parser.py:
```diff
@@ -13,8 +15,8 @@
-filename = "data/spacerdatabase.txt" # File from CRISPRdb to sort 
-spacer_db = open("%s" %(filename),"r") 
+filename = "data/spacerdatabase.txt"  # File from CRISPRdb to sort
+spacer_db = open("%s" % filename, "r")
```
Sure, I'll remove it and replace with `open(filename, 'r')`.

delirious-lettuce commented on parserscripts/extract_CRISPRdb.py:
```diff
@@ -21,49 +24,59 @@ def get_taxons_from_CRISPRdb():
+                           .replace('<br/n/>', ''))
+        crispr_id_value = crispr_id_value.replace('\n', '')
+        crispr_id_value = crispr_id_value.replace('\t', '')
+        crispr_id_value = crispr_id_value.replace('<br />', '')
```
Are you sure? I chained that one above because there were so many dots, these ones only have one dot...?

delirious-lettuce replied:
@QuLogic ,

Do you mean like this?

```python
        crispr_id_value = (cell('font')[0]
                           .content
                           .encode('ascii', 'ignore')
                           .replace('<br/n/>', '')
                           .replace('\n', '')
                           .replace('\t', '')
                           .replace('<br />', ''))
```

delirious-lettuce commented on parserscripts/jointable.py:
```diff
@@ -1,89 +1,111 @@
-	conn.close()
-	return r
+def read_repeat_file(filename):
+    fin = open(filename, 'r')
```
Sure, no problem.

delirious-lettuce commented:
@QuLogic ,

Thanks for the detailed review!

Sorry about not using multiple commits, I will remember that for the future. It's kind of funny because I sent a PR last year to someone and they complained that I used too many commits! :confused:

QuLogic commented on parserscripts/extract_CRISPRdb.py:
```diff
@@ -21,49 +24,59 @@ def get_taxons_from_CRISPRdb():
-        crispr_id_value = crispr_id_value.replace('<br />','')
+        crispr_id_value = (cell('font')[0]
+                           .content
+                           .encode('ascii', 'ignore')
```
@mbonsma do you use this script any more? This doesn't look right. `encode` is `str` -> `bytes` and the next method is working with strings.

Also, `pattern.web` seems broken on Python 3.

mbonsma replied:
If we're going to run CRISPRfinder locally, we don't need the scraping part of the script but we'd still need the part that parses the table. (It addresses #35.) 

I'll make a new issue to look this one over.

QuLogic commented:
It's always easier to squash small commits than split large commits.

mbonsma commented:
This all looks good to me! Thanks @QuLogic and @delirious-lettuce!

delirious-lettuce commented:
@mbonsma ,

I'm glad I could help out (even in a small way), thanks for your time!
